server/data/interfaces/referenceData.ts

```diff
@@ -0,0 +1,13 @@
+import { CorePersonRecordReferenceDataDomain } from './personIntegrationApi/personIntegrationApiClient'
+import { HealthAndMedicationReferenceDataDomain } from './healthAndMedicationApi/healthAndMedicationApiClient'
+
+export interface ReferenceDataCodeDto {
+  id: string
+  code: string
+  description: string
+  listSequence: number
+  isActive: boolean
+  parentCode?: string
+}
```

Comment:
This is the generic data class that we'll use in the profile and map other dtos from the various APIs to